processing/src/main/java/io/druid/query/groupby/GroupByQueryHelper.java

```diff
@@ -111,6 +114,13 @@ public String apply(DimensionSpec input)
       );
     }
 
+    Map<String, ValueType> typeHints = BaseQuery.getContextTypeHints(query);
+    if (typeHints != null) {
+      for (String dimName : typeHints.keySet()) {
+        index.loadDimensionCapability(dimName, typeHints.get(dimName));
```

Comment:
"load" doesn't really evoke in me the idea of "hey index-thingie, consider this dim as having these capabilities please".  How about `applyDimensionCapability` or `specifyDimensionCapability` or `enforceDimensionCapability`?